api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java

```diff
@@ -2047,14 +2054,19 @@ public CohortDefinition getMQC12P2DEN(String indicatorFlag) {
     compositionCohortDefinition.addSearch("F", EptsReportUtils.map(transfOut, MAPPING1));
 
     if (indicatorFlag.equals("A") || indicatorFlag.equals("C")) {
-      compositionCohortDefinition.setCompositionString("(A AND B1) NOT (C OR D OR E)");
+      compositionCohortDefinition.setCompositionString(
+          "(A AND B1) AND  NOT B1E  AND  NOT  (C OR D OR F)");
     }
     if (indicatorFlag.equals("B") || indicatorFlag.equals("D")) {
-      compositionCohortDefinition.setCompositionString("(A AND B2) AND NOT (C OR D OR E)");
+      compositionCohortDefinition.setCompositionString(
+          "(A AND B2) AND  NOT B2E  AND  NOT  (C OR D OR F)");
     }
     if (indicatorFlag.equals("E")) {
       compositionCohortDefinition.setCompositionString("(A AND B1 AND C) AND NOT (D OR E)");
```

Comment:
This needs a small update according to the spec

Reply:
done